src/schema/v2/me/user_flags.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  GraphQLError,
+  GraphQLFieldConfig,
+  GraphQLScalarType,
+  Kind,
+  ObjectValueNode,
+  print,
+  ValueNode,
+} from "graphql"
+import Maybe from "graphql/tsutils/Maybe"
+import { camelCase } from "lodash"
+import { ResolverContext } from "types/graphql"
+
+export const UserFlagsType = new GraphQLScalarType({
+  name: "UserFlagsType",
+  description:
+    "The user's user_flags. Values of keys in user_flags can be anything and keys may not be known beforehand.",
+  serialize: (value) => value,
+  parseValue: (value) => value,
+  parseLiteral: (ast, variables) => {
+    if (ast.kind !== Kind.OBJECT) {
+      throw new TypeError(
+        `Object cannot represent non-object value: ${print(ast)}`
+      )
+    }
+
+    return parseObject("Object", ast, variables)
+  },
+})
+
+export const userFlagsResolver = async (root, options, loaders) => {
+  const { collectorProfileLoader } = loaders
+  if (!collectorProfileLoader) {
+    throw new Error("You need to be signed in to perform this action")
+  }
+
+  const otherFlags: { [key: string]: any } = {}
+
+  const result = await collectorProfileLoader(options)
+  const { bio, icon, other_relevant_positions } = result
+  const profileIsComplete =
+    root.profession && bio && icon && other_relevant_positions
+
+  if (!profileIsComplete) {
+    otherFlags.collector_profile_incomplete_at = new Date().toISOString()
+  }
+
+  const mergedResult = {
+    ...root.user_flags,
+    ...otherFlags,
+  }
+
+  const res = {}
+  const keys = Object.keys(mergedResult)
+  keys.forEach((key) => {
+    res[camelCase(key)] = mergedResult[key]
```

Comment:
same comment here as above the snake case, I think we don't need this

Reply:
@MounirDhahri Similar to my comment above. This ensures that a standard json camelized property is always applied to the keys we get back. We can always expect that rails will return these values as snakecase as that by convention is how hash_keys are used. But in frontend clients we expect the keys camelized

Reply:
thanks for the explanation Kizito